### DISCUSSION TITLE
"Failed to look up crate package" for workspace crates

### DISCUSSION THREAD

**Discussion by Turbo87** [renovatebot/renovate#33148] [Request Help] [answered]
### How are you running Renovate?

A Mend.io-hosted app

### Please tell us more about your question or problem

> [!WARNING]
> Renovate failed to look up the following dependencies: `Failed to look up crate package crates_io_cdn_logs`, `Failed to look up crate package crates_io_database`, `Failed to look up crate package crates_io_database_dump`, `Failed to look up crate package crates_io_env_vars`, `Failed to look up crate package crates_io_github`, `Failed to look up crate package crates_io_index`, `Failed to look up crate package crates_io_markdown`, `Failed to look up crate package crates_io_pagerduty`, `Failed to look up crate package crates_io_tarball`, `Failed to look up crate package crates_io_team_repo`, `Failed to look up crate package crates_io_worker`, `Failed to look up crate package crates_io_test_db`.
> 
> Files affected: `Cargo.toml`, `crates/crates_io_database/Cargo.toml`, `crates/crates_io_database_dump/Cargo.toml`, `crates/crates_io_index/Cargo.toml`, `crates/crates_io_smoke_test/Cargo.toml`, `crates/crates_io_test_db/Cargo.toml`, `crates/crates_io_worker/Cargo.toml`

but these crates are workspace-internal packages/dependencies, so renovatebot would not find them on the registry. This is the first time I've seen this warning, so it seems likely to be caused by a recent change?

### Logs (if relevant)

see https://developer.mend.io/github/rust-lang/crates.io/-/job/0193cffc-1761-7268-964b-6771a7233971, but I didn't spot anything particularly relevant in them

**Comment by rarkins** [comment]
Example internal reference: https://github.com/rust-lang/crates.io/blob/f96279622d751e311ef3cb3c37903bba3981fca7/crates/crates_io_database_dump/Cargo.toml#L24

Related logs:

```
DEBUG: GET https://raw.githubusercontent.com/rust-lang/crates.io-index/master/cr/at/crates_io_test_db = (code=ERR_NON_2XX_3XX_RESPONSE, statusCode=404 retryCount=0, duration=5756)
DEBUG: Datasource 404
{
  "datasource": "crate"
  "packageName": "crates_io_test_db"
  "url": "https://raw.githubusercontent.com/rust-lang/crates.io-index/master/cr/at/crates_io_test_db"
}

DEBUG: Failed to look up crate package crates_io_test_db
{
  "dependency": "crates_io_test_db"
  "packageFile": "Cargo.toml"
}
```

Then:

```
          {
            "currentValue": "",
            "datasource": "crate",
            "depName": "crates_io_test_db",
            "depType": "dev-dependencies",
            "lockedVersion": "0.0.0",
            "managerData": {
              "nestedVersion": false
            },
            "packageName": "crates_io_test_db",
            "versioning": "cargo",
            "warnings": [
              {
                "topic": "crates_io_test_db",
                "message": "Failed to look up crate package crates_io_test_db"
              }
            ],
            "updates": []
          },
```

**Reply by rarkins** [reply]
Running with --dry-run=extract I get:

```
               {
                 "currentValue": "",
                 "managerData": {"nestedVersion": false},
                 "datasource": "crate",
                 "skipReason": "path-dependency",
                 "depName": "crates_io_test_db",
                 "depType": "dev-dependencies",
                 "lockedVersion": "0.0.0"
               },
```

I'm not sure why yet, but I'm pretty sure that the problem is related to my `skipReason` change: https://github.com/renovatebot/renovate/pull/33108/files#diff-a8a085c9d3130eebaeaf7348942bfce3ae726c9cdc5ef3ec0a94925d75b76806R131

**Comment by rarkins** [comment, marked as answer]
Converted to p1 bug: https://github.com/renovatebot/renovate/issues/33171
PR fix: https://github.com/renovatebot/renovate/pull/33172

**Reply by Turbo87** [reply]
awesome, thank you! :)
